docs/exposing_new_functions.mld

```diff
@@ -0,0 +1,31 @@
+{0 Exposing New Functions to stanc3}
+
+
+
+{1 Background}
+
+For a function to be built into Stan, it has to be included in the Stan Math library and its signature has to be
+exposed to the parser. To do the latter, we have to add a corresponding line in [src/middle/Stan_math_signatures.ml].
+The parser uses the signatures defined there to do the type checking. 
+
+
+{1 Adding a distribution function }
+To add a distribution, we have to find the line containing [let distributions =]. The existing distributions can be
+used for reference. The first argument defines the type of function we want to add.
+The second argument is the function name without the [_...] part. The third argument specifies
+the argument types of the function. The last argument describes whether the return type can be represented as an
```

Comment:
It's worth describing what these types are a bit more. For example, the thing you use in "foo" (`DVReal`) can be a real, array[] real, row_vector, or vector (`V` stands for "vectorized")

Reply:
I'd be happy to help adding some more details, I'll see if I can understand what's going on and expand this a bit.